universal-login-react/src/ui/TopUp/TopUp.tsx

```diff
@@ -31,69 +34,103 @@ export interface TopUpProps {
 export const TopUp = ({walletService, startModal, modalClassName, hideModal, isModal, logoColor}: TopUpProps) => {
   const [modal, setModal] = useState<TopUpComponentType>(startModal || TopUpComponentType.choose);
   const [amount, setAmount] = useState('');
-
+  const [paymentMethod, setPaymentMethod] = useState<TopUpProvider | undefined>(undefined);
   const relayerConfig = walletService.sdk.getRelayerConfig();
   const contractAddress = walletService.getContractAddress();
+  const [topUpMethod, setTopUpMethod] = useState<TopUpMethod>(undefined);
 
   const onPayClick = (provider: TopUpProvider, amount: string) => {
+    setTopUpMethod(undefined);
     setModal(toTopUpComponentType(provider));
     setAmount(amount);
   };
 
   const getTopUpMethodChooser = () => (
     <ChooseTopUpMethod
-      walletService={walletService}
-      onPayClick={onPayClick}
-      logoColor={logoColor}
+      topUpMethod={topUpMethod}
+      setTopUpMethod={setTopUpMethod}
     />
   );
 
-  if (!relayerConfig) {
-    return <Spinner />;
-  } else if (modal === TopUpComponentType.choose) {
-    if (isModal) {
-      return <ModalWrapper message={walletService.sdk.getNotice()} modalClassName="top-up-modal" hideModal={hideModal}>{getTopUpMethodChooser()}</ModalWrapper>;
-    }
-    return getTopUpMethodChooser();
-  } else if (modal === TopUpComponentType.safello) {
-    return (
-      <ModalWrapper modalClassName={modalClassName} hideModal={() => setModal(TopUpComponentType.choose)}>
+  const renderTopUpContent = () => {
+    if (!relayerConfig) {
+      return <Spinner />;
+    } else if (topUpMethod === 'fiat') {
+      return (
+        <>
+          {getTopUpMethodChooser()}
```

Comment:
For me inlined fuction will be more readable
```suggestion
    <ChooseTopUpMethod topUpMethod={topUpMethod} setTopUpMethod={setTopUpMethod} />
```